Coding/fw/src/CANAdapter.h

```diff
@@ -0,0 +1,259 @@
+    /** 
+    * Configures and starts the CAN Controller to use the user values.
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t begin()
+    {
+        uint8_t isError = 0;
+        if (CAN.begin(500E3) == 0)      /**< Starts CAN channel with 500kbps Baudrate */
+        {
+            isError = 1;
+        }
+        else
+        {
+            CAN.sleep();
+        }
+
+        return isError;
+    }
+
+    /** 
+    * Stops the Controller Module without destroying the instance.
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t end()
+    {
+        CAN.end();
+        return 0;
+    }
+
+    /**
+    * Send a Frame from Serial to CAN Channel
+    * @param &Frame     Reference to the Frame to be sended
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t send(const Frame &frame)
+    {
+        uint8_t isError = 0;
+        if (frame.m_extended)
+        {
+            if (CAN.beginExtendedPacket(frame.m_id, frame.m_dlc, frame.m_rtr) == 0)
+            {
+                isError = 1;
+            }
+        }
+        else
+        {
+            if (CAN.beginPacket(frame.m_id, frame.m_dlc, frame.m_rtr) == 0)
+            {
+                isError = 1;
+            }
+        }
+
+        CAN.write(frame.m_data, frame.m_dlc);
+
+        if (CAN.endPacket() == 0)
+        {
+            isError = 1;
+        }
+
+        return isError;
+    }
+
+    /**
+    * Set the State of the CAN Channel.
+    * @param state          BUS_STATE to be set to the CAN Channel
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t setState(BUS_STATE state)
+    {
+        uint8_t isError = 0;
+        switch (state)
+        {
+        case CLOSED:
+            CAN.sleep();
+            m_currentState = CLOSED;
+            break;
+
+        case NORMAL:
+            if (m_baudRate == 0)
+            {
+                isError = 1;
+            }
+            CAN.wakeup();
+            m_currentState = NORMAL;
+            break;
+
+        case LISTEN_ONLY:
+            if (m_baudRate == 0)
+            {
+                isError = 1;
+            }
+            CAN.wakeup();
+            m_currentState = LISTEN_ONLY;
+            break;
+
+        default:
+            isError = 1;
+            break;
+        }
+
+        return isError;
+    }
+
+    /**
+    * Set the Baudrate of the CAN Channel.
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t setBaudrate(uint32_t baudrate)
+    {
+        uint8_t isError = 0;
+        m_baudRate = baudrate;
+        CAN.end();
+        if (CAN.begin(m_baudRate) == 0)
+        {
+            isError = 1;
+        }
+        CAN.sleep();
+        return isError;
+    }
+
+    /**
+    * Sent the BTR Registers of the CAN Channel.
+    * @return 0 for OK, 1 for Error 
+    */
+    uint8_t setBTR(uint8_t btr0, uint8_t btr1)
+    {
+        return 1; /**< Must write to register. It returns error as the Controller does not allow it */
```

Comment:
Add a /* TODO: */ comment for things that are not yet implemented. This makes it easy to find such missing things later.